dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/SdkFeature.kt

```diff
@@ -123,11 +123,11 @@ internal class SdkFeature(
 
     // region FeatureScope
 
-    // TODO RUMM-0000 there is no thread switch here, it stays the same.
-    // Need to clarify the threading. We either switch thread here or in Storage.
-    // Or give the ability to specify the executor to the caller.
-    @Suppress("ThreadSafety")
     override fun withWriteContext(callback: (DatadogContext, EventBatchWriter) -> Unit) {
+        // TODO RUMM-0000 thread safety. Thread switch happens in Storage right now. Open questions:
+        // * what if called wants to have a sync operation, without thread switch
```

Comment:
I am thinking when will be the case/need for that. Do you have some situations in mind ?

Reply:
Crash reporting - we need write it as log or RUM event in a blocking manner.

Reply:
true, forgot about that.